cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -729,7 +729,10 @@ struct set_final_row_count {
     if (i < pages.size() - 1 && (pages[i + 1].chunk_idx == page.chunk_idx)) { return; }
     size_t const page_start_row = chunk.start_row + page.chunk_row;
     size_t const chunk_last_row = chunk.start_row + chunk.num_rows;
-    page.num_rows               = chunk_last_row - page_start_row;
+    // Mark `is_num_rows_adjusted` to signal string decoders that the `num_rows` of this page has
+    // been adjusted.
+    page.is_num_rows_adjusted = page.num_rows != (chunk_last_row - page_start_row);
```

Comment:
is this always executed? any cases where we need to set `is_num_rows_adjusted` to false?

Reply:
Nope, only need to set it `true` for the last page(s).

Reply:
Done in https://github.com/rapidsai/cudf/pull/17321/commits/241a88f39d20d68cf12a9f84e3ad4762dec27379